modules/nlu/src/backend/language/language-provider.ts

```diff
@@ -135,6 +150,56 @@ export class RemoteLanguageProvider implements LanguageProvider {
     return this
   }
 
+  private extractLangServerVersion(data) {
+    const version = semver.valid(semver.coerce(data.version))
+
+    if (!version) {
+      throw new Error('Lang server has an invalid version')
+    }
+    this._state.langServerVersion = semver.clean(version)
+  }
+
+  private computeCacheFilesPaths = () => {
+    const versionHash = this.computeVersionHash()
+    this._vectorsCachePath = path.join(this._cacheDir, `${VECTOR_FILE_PREFIX}_${versionHash}.json`)
+    this._junkwordsCachePath = path.join(this._cacheDir, `${JUNK_FILE_PREFIX}_${versionHash}.json`)
+    this._tokensCachePath = path.join(this._cacheDir, `${TOKEN_FILE_PREFIX}_${versionHash}.json`)
+  }
+
+  private clearOldCacheFiles = async () => {
+    const cacheExists = await fse.pathExists(this._cacheDir)
+    if (!cacheExists) {
+      return
+    }
+
+    const allCacheFiles = await fse.readdir(this._cacheDir)
+
+    const currentHash = this.computeVersionHash()
+
+    const fileStartWithPrefix = (fileName: string) => {
+      return (
+        fileName.startsWith(VECTOR_FILE_PREFIX) ||
+        fileName.startsWith(TOKEN_FILE_PREFIX) ||
+        fileName.startsWith(JUNK_FILE_PREFIX)
+      )
+    }
+
+    const fileEndsWithIncorrectHash = (fileName: string) => {
```

Comment:
that's quite well written, it could have been as dumb as

```
const fileEndsWithIncorrectHash = (fileName: string) => !fileName.includes(currentHash)
```

Reply:
yes, you're right